library/src/hcc_detail/hipsolver.cpp

```diff
@@ -697,6 +714,219 @@ catch(...)
     return exception2hip_status();
 }
 
+/******************** GESVD ********************/
+hipsolverStatus_t hipsolverSgesvd_bufferSize(hipsolverHandle_t handle, int m, int n, int* lwork)
+try
+{
+    *lwork = 0;
+    return HIPSOLVER_STATUS_NOT_SUPPORTED;
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverDgesvd_bufferSize(hipsolverHandle_t handle, int m, int n, int* lwork)
+try
+{
+    *lwork = 0;
+    return HIPSOLVER_STATUS_NOT_SUPPORTED;
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverCgesvd_bufferSize(hipsolverHandle_t handle, int m, int n, int* lwork)
+try
+{
+    *lwork = 0;
+    return HIPSOLVER_STATUS_NOT_SUPPORTED;
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverZgesvd_bufferSize(hipsolverHandle_t handle, int m, int n, int* lwork)
+try
+{
+    *lwork = 0;
+    return HIPSOLVER_STATUS_NOT_SUPPORTED;
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverSgesvd(hipsolverHandle_t handle,
+                                  signed char       jobu,
+                                  signed char       jobv,
+                                  int               m,
+                                  int               n,
+                                  float*            A,
+                                  int               lda,
+                                  float*            S,
+                                  float*            U,
+                                  int               ldu,
+                                  float*            V,
+                                  int               ldv,
+                                  float*            work,
+                                  int               lwork,
+                                  float*            rwork,
+                                  int*              devInfo)
+try
+{
+    if(!rocblas_is_managing_device_memory((rocblas_handle)handle))
+        rocblas_set_workspace((rocblas_handle)handle, nullptr, 0);
+
+    return rocblas2hip_status(rocsolver_sgesvd((rocblas_handle)handle,
+                                               char2rocblas_svect(jobu),
+                                               char2rocblas_svect(jobv),
+                                               m,
+                                               n,
+                                               A,
+                                               lda,
+                                               S,
+                                               U,
+                                               ldu,
+                                               V,
+                                               ldv,
+                                               rwork,
+                                               rocblas_inplace,
```

Comment:
Using in-place should be, in most cases, slower than out-of-place...
So, there are more chances for this function on the HIP side to be always slower than on CUDA side. I would say always use out-of-place instead, but we don't want to require too much more memory on HIP than on CUDA. So I suggest you check, for some given sizes, the memory required by cuSOLVER and rocSOLVER in both cases, in-place and out-of-place. If out-of-place is compared to what cuSOLVER needs, then switch to always compute out-of-place.

Reply:
I seem to recall having a reason for making it in-place, but I've forgotten what it was. I'll try switching it over to out-of-place and comparing memory sizes, and perhaps I'll remember what my reasoning was.